src/main/java/ru/sbt/mipt/oop/StandardProcessingScript.java

```diff
@@ -0,0 +1,32 @@
+package ru.sbt.mipt.oop;
+
+import static ru.sbt.mipt.oop.SensorEventType.*;
+
+public class StandardProcessingScript implements ProcessingScript {
+
+    public void doScript(SmartHome smartHome, Sensor sensor) {
+        SensorEvent event = sensor.getNextSensorEvent();
+        while (event != null) {
+            System.out.println("Got event: " + event);
+            if (event.getType() == LIGHT_ON || event.getType() == LIGHT_OFF) {
+                // событие от источника света
+                EventProcessor processor = new LightEventProcessor(smartHome, event);
+                processor.processEvent();
+            }
+            if (event.getType() == DOOR_OPEN || event.getType() == DOOR_CLOSED) {
+                // событие от двери
+                EventProcessor processor = new DoorEventProcessor(smartHome, event);
+                processor.processEvent();
+
+                // если мы получили событие о закрытие двери в холле - это значит, что была закрыта входная дверь.
+                // в этом случае мы хотим автоматически выключить свет во всем доме (это же умный дом!)
+                Room room = smartHome.getRoomByDoor(event.getObjectId());
+                if (room != null && event.getType() == DOOR_CLOSED && room.getName().equals("hall")) {
```

Comment:
https://github.com/iagoshkov/smart-home-2020/blob/master/STUDENT_GUIDE.md#%D1%80%D0%B0%D0%B7%D0%B4%D0%B5%D0%BB%D1%8F%D0%B9%D1%82%D0%B5-%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%8A%D0%B5%D0%BA%D1%82%D0%BE%D0%B2-%D0%B8-%D1%81%D1%86%D0%B5%D0%BD%D0%B0%D1%80%D0%B8%D0%B8-%D1%83%D0%BC%D0%BD%D0%BE%D0%B3%D0%BE-%D0%B4%D0%BE%D0%BC%D0%B0